src/Tribe/Support/FIlter_Checker_Report.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Assembles a report of recently updated plugin views and template overrides in
+ * possible revision, for each plugin that registers itself and its template
+ * filepaths.
+ */
+class Tribe__Support__Filter_Checker_Report {
+	/**
+	 * Contains the individual reports for each registered plugin.
+	 *
+	 * @var array
+	 */
+	protected static $plugin_reports = [];
+
+	/**
+	 * Container for finished report.
+	 *
+	 * @var string
+	 */
+	protected static $complete_report = '';
+
+	/**
+	 * Provides an up-to-date report concerning filter overrides.
+	 *
+	 * @return string
+	 */
+	public static function generate() {
+		foreach ( self::registered_filters() as $plugin_name => $plugin_filters ) {
+			self::generate_for( $plugin_name, $plugin_filters );
+		}
+
+		self::wrap_report();
+		return self::$complete_report;
+	}
+
+	protected static function registered_filters() {
+		/**
+		 * Provides a mechanism for plugins to register information about their filters.
+		 *
+		 * This should be done by adding an entry to $plugin_filters where the key
+		 * should be the plugin name and the value an array of filter slugs to check:
+		 *
+		 *     plugin_name => [
+		 *         filter_slug,
+		 *         another_filter_slug,
+		 * 		],
+		 *
+		 * @var array $plugin_filters
+		 */
+		return apply_filters(
+			'tribe_support_registered_filter_overrides',
+			[
+				'the-events-calendar' => [
```

Comment:
This seems a maintenance nightmare that will either end up with this being abandoned or this code never being used.
Why not make this dynamic and collect all the stuff attached to any filter starting with `tribe_`; that is consistent in our code and would not require maintenance.

Reply:
That's the idea - I just put in some dummy data for now to illustrate and test.

Reply:
We also know that our plugins themselves extend each other, so we wouldn't want to count that as an "override" from the customer perspective.